pkg/apiserver/metrics/metrics.go

```diff
@@ -42,7 +42,7 @@ var (
 			// Use buckets ranging from 125 ms to 8 seconds.
 			Buckets: prometheus.ExponentialBuckets(125000, 2.0, 7),
 		},
-		[]string{"verb", "resource"},
+		[]string{"verb", "resource", "client"},
```

Comment:
These values stay around forever - so it's a very easy cluster DoS (at
least the last time I tested it).


On Nov 19, 2015, at 3:56 PM, Timothy St. Clair <notifications@github.com>
wrote:

In pkg/apiserver/metrics/metrics.go
<https://github.com/kubernetes/kubernetes/pull/17502#discussion_r45398937>:

> @@ -42,7 +42,7 @@ var (
>  			// Use buckets ranging from 125 ms to 8 seconds.
>  			Buckets: prometheus.ExponentialBuckets(125000, 2.0, 7),
>  		},
> -		[]string{"verb", "resource"},
> +		[]string{"verb", "resource", "client"},

I'm not terribly concerned about how much memory, as we load-balance at
higher levels, and there are caps on the number of connections.

I guess a better question is, what is your objective in tooling per-client.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/17502/files#r45398937>.